volatility3/framework/plugins/linux/ptrace.py

```diff
@@ -0,0 +1,120 @@
+from volatility3.framework.configuration import requirements
+from volatility3.framework.interfaces import plugins
+from volatility3.plugins.linux import pslist
+
+vollog = logging.getLogger(__name__)
+
+
+class Ptrace(plugins.PluginInterface):
+    """Enumerates tracer and tracee tasks"""
+
+    _required_framework_version = (2, 10, 0)
+    _version = (1, 0, 0)
+
+    @classmethod
+    def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]:
+        return [
+            requirements.ModuleRequirement(
+                name="kernel",
+                description="Linux kernel",
+                architectures=LINUX_ARCHS,
+            ),
+            requirements.PluginRequirement(
+                name="pslist", plugin=pslist.PsList, version=(2, 2, 1)
+            ),
+        ]
+
+    @classmethod
+    def enumerate_ptraced_tasks(
+        cls,
+        context: interfaces.context.ContextInterface,
+        symbol_table: str,
+    ):
+        vmlinux = context.modules[symbol_table]
+
+        tsk_struct_symname = vmlinux.symbol_table_name + constants.BANG + "task_struct"
+
+        tasks = pslist.PsList.list_tasks(
+            context,
+            symbol_table,
+            filter_func=pslist.PsList.create_pid_filter(),
+            include_threads=True,
+        )
+
+        for task in tasks:
+            tracing_tid_list = [
+                int(task_being_traced.pid)
+                for task_being_traced in task.ptraced.to_list(
+                    tsk_struct_symname, "ptrace_entry"
+                )
+            ]
+
+            if task.ptrace == 0 and not tracing_tid_list:
+                continue
+
+            flags = (
+                PT_FLAGS(task.ptrace).flags
+                if task.ptrace != 0
+                else renderers.NotAvailableValue()
+            )
+
+            traced_by_tid = (
+                task.parent.pid
+                if task.real_parent != task.parent
+                else renderers.NotAvailableValue()
+            )
+
+            tracing_tids = ",".join(map(str, tracing_tid_list))
+
+            yield task.comm, task.tgid, task.pid, traced_by_tid, tracing_tids, flags
+
+    def _generator(self, symbol_table):
+        for fields in self.enumerate_ptraced_tasks(self.context, symbol_table):
+            yield (0, fields)
+
+    @staticmethod
+    def format_fields_with_headers(headers, generator):
+        """Uses the headers type to cast the fields obtained from the generator"""
+        for level, fields in generator:
+            formatted_fields = []
+            for header, field in zip(headers, fields):
+                header_type = header[1]
+
+                if isinstance(
+                    field, (header_type, interfaces.renderers.BaseAbsentValue)
+                ):
+                    formatted_field = field
+                elif isinstance(field, objects.Array) and header_type is str:
+                    formatted_field = utility.array_to_string(field)
```

Comment:
This whole method is trying to circumvent the restrictions rather than working within them.  Don't pass arrays through the field, pass individual numbers with repeated data instead.  There really shouldn't be any need to reformat the response from the generator before feeding it back out.

Reply:
The array is `task.comm`

Reply:
Ok, sorry, I misunderstood.  I thought this was for something that was a set of numbers like the `tracing_tid_list`.  Why wouldn't we do the conversion to string during the output, rather than yielding an array?  We already know which field is going to be returned as an array, it would lose volatility information (like its layer and offset) but it's likely to be far more usable?  It's also strange that *knowing* the field isn't an instance of `header_type`, you apply it like a cast?  What situations is that trying to catch?